### ISSUE TITLE
Allow JWT in query string per decorated method

### ISSUE THREAD

**Issue by dmulter** [vimalloc/flask-jwt-extended#370]
It's very convenient to use `JWT_TOKEN_LOCATION` for specifying the expected location(s) of the token, but that is for all methods. In my use case, I would like to offer only select endpoints the ability to pass the token via a query parameter. This way all the endpoints require the token in the header for security reasons, but one type of endpoint (/help) can be used easily in a browser with GET and the token in the URL.

This could be done with separate decorators for required/optional JWTs, using a method-specific handler to help in the extraction, etc.

**Comment by vimalloc-mavenlink** [comment]
I worked on something similar before, but ended up scrapping it because it would have caused a breaking change to the decorators. However I have already made the breaking change to add arguments to the decorators as part of the 4.0.0 branch, so I think it would be easy enough to implement now. If you want to take a stab at it I would love a PR, I think it will just involve passing in a `location` kwarg to the decorator that is then forwarded to https://github.com/vimalloc/flask-jwt-extended/blob/4.0.0-dev/flask_jwt_extended/view_decorators.py#L199, and defaults to the config locations if one is not explicitly passed in.

Otherwise I would be happy to add that when time allows, but it might be a while. Got a lot of unrelated stuff on my plate at the moment.
